### PR TITLE
Compression, share, meta, title changes

### PR DESCRIPTION
**Pull request by shreyagarwaldev**

### CODE REVIEW

File: src/app/contact-us/contact-us.component.ts [hunk 1]
```diff
@@ -53,9 +54,27 @@ export class ContactUsComponent {
   constructor(
     private http:Http,
     private globalConstants: GlobalConstantsRepository,
-    private gaService: GoogleAnalyticsService) {
+    private gaService: GoogleAnalyticsService,
+    title: Title,
```
Comment (sugandhadev) [inline src/app/contact-us/contact-us.component.ts]: private?

File: src/app/navbar/navbar.component.html [hunk 2]
```diff
@@ -30,7 +30,7 @@
 </div>
 <div class="flyout-blocks social" [hidden]="hideShare">
   <div class="social-icons">
-    <i class="fa fa-facebook-official" aria-hidden="true"></i>
-    <i class="fa fa-twitter-square" aria-hidden="true"></i>
+    <div [attr.data-href]="router.url"><a style="color:inherit" title="Share on Facebook" class="fb-xfbml-parse-ignore" target="_blank" href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.thepixelatedplanet.com%2F&amp;src=sdkpreparse"><i class="fa fa-facebook-official" aria-hidden="true"></i></a></div>
```
Comment (sugandhadev) [inline src/app/navbar/navbar.component.html]: inline, adding here to keep this in mind

File: src/app/navbar/navbar.component.ts [hunk 3]
```diff
@@ -37,8 +43,17 @@ export class NavComponent {
...
 
     submitEmail() {
-        console.log(this.emailId);
-        this.toggleSubscribeBlock();
+        let headers = new Headers();
+        headers.append('Content-Type', 'application/json');
+        let options = new RequestOptions();
+        options.headers = headers;
+        this.http.post(this.globalConstants.getSubscribeAPIUrl(), `"${this.emailId}"`, options).toPromise().then(e => {
+            this.toggleSubscribeBlock();
```
Comment (sugandhadev) [inline src/app/navbar/navbar.component.ts:54]: we can move this call outside the http call

File: src/app/workshop-details/workshop-details.component.html [hunk 4]
```diff
@@ -63,13 +63,20 @@ <h2 class="highlight-headings">Price</h2>
                 <h2 class="highlight-headings">Workshop Link(s)</h2>
                 <div class="dates-wrapper">
                     <div class="dates-information" *ngFor="let detail of workshopDetails.multiWorkshopDetails;">
-                        <a href="{{detail.externalLink}}" (click)="clickExternalLink(detail.link)">
+                        <a title="Details for workshop starting on {{detail.startDateStr}}" href="{{detail.externalLink}}" (click)="clickExternalLink(detail.link)">
```
Comment (sugandhadev) [inline src/app/workshop-details/workshop-details.component.html:66]: is this working?

File: src/app/workshop-details/workshop-details.component.html [hunk 5]
```diff
@@ -63,13 +63,20 @@ <h2 class="highlight-headings">Price</h2>
...
                             <i class="fa fa-calendar"></i>
                             <span>{{detail.startDateStr}} - {{detail.endDateStr}}</span>
                         </a>
                     </div>
                 </div>
             </div>
+            <div class="wsd-share">
```
Comment (sugandhadev) [inline src/app/workshop-details/workshop-details.component.html:73]: why did you add this?

File: src/app/workshop-details/workshop-details.component.html [hunk 6]
```diff
@@ -81,8 +88,8 @@ <h2 class="highlight-headings">Workshop Link(s)</h2>
                 </div>
             </div>
 
-            <a class="prev" (click)="plusSlides(-1)">&#10094;</a>
-            <a class="next" (click)="plusSlides(1)">&#10095;</a>
+            <a title="Go to previous image" class="prev" (click)="plusSlides(-1)">&#10094;</a>
```
Comment (sugandhadev) [inline src/app/workshop-details/workshop-details.component.html:89]: go to previous photography workshop image?

File: src/app/workshops-list/workshops-list.component.html [hunk 7]
```diff
@@ -24,11 +24,11 @@ <h1 class="name truncate" title="{{ data.name }}">
             </a>
         </article>
         <div *ngIf="itemsPerPage > 4" class="ws-pages">
-            <a class="page-numbers" *ngIf="pageNumbers" [ngClass]="{'disabled': activePage <=1 }" [href]="createPageLink(activePage-1)">
+            <a title="Go to previous page" class="page-numbers" *ngIf="pageNumbers" [ngClass]="{'disabled': activePage <=1 }" [href]="createPageLink(activePage-1)">
```
Comment (sugandhadev) [inline src/app/workshops-list/workshops-list.component.html:27]: same as above?

**Review by sugandhadev — CHANGES_REQUESTED** [review]

File: src/app/workshop-details/workshop-details.component.html (hunk 5 above)
Comment (sugandhadev) [reply src/app/workshop-details/workshop-details.component.html:73]: why couldn't we just show the navbar one and pass the url which we want?

**Review by sugandhadev — APPROVED** [review]

File: src/app/contact-us/contact-us.component.ts (hunk 1 above)
Comment (shreyagarwaldev) [reply src/app/contact-us/contact-us.component.ts]: can make it public when needed

File: src/app/navbar/navbar.component.ts (hunk 3 above)
Comment (shreyagarwaldev) [reply src/app/navbar/navbar.component.ts:54]: the correct thing to do is to show an error and not toggle if it fails. That's why I moved inside. Let's keep it like this and start showing error soon

File: src/app/workshop-details/workshop-details.component.html (hunk 4 above)
Comment (shreyagarwaldev) [reply src/app/workshop-details/workshop-details.component.html:66]: of course

File: src/app/workshop-details/workshop-details.component.html (hunk 5 above)
Comment (shreyagarwaldev) [reply src/app/workshop-details/workshop-details.component.html:73]: discussed offline

File: src/app/workshop-details/workshop-details.component.html (hunk 6 above)
Comment (shreyagarwaldev) [reply src/app/workshop-details/workshop-details.component.html:89]: these are not used for SEO through text.. just used for proving usability of website
